projects/packages/forms/src/blocks/field-file/edit.js

```diff
@@ -0,0 +1,99 @@
+import { getJetpackExtensionAvailability } from '@automattic/jetpack-shared-extension-utils';
+import {
+	useBlockProps,
+	useInnerBlocksProps,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import JetpackFieldControls from '../shared/components/jetpack-field-controls';
+import { UpsellNudge } from '../shared/components/upsell-nudge';
+import useFormWrapper from '../shared/hooks/use-form-wrapper';
+import useJetpackFieldStyles from '../shared/hooks/use-jetpack-field-styles';
+import useParentFormClientId from '../shared/hooks/use-parent-form-client-id';
+import './editor.scss';
+
+export default function FileFieldEdit( props ) {
+	const { attributes, clientId, isSelected, setAttributes, name, className } = props;
+	const { id, required, width } = attributes;
+
+	const fieldFileAvailability = getJetpackExtensionAvailability( 'field-file' );
```

Comment:
Maybe it's not an issue, but this won't be reactive (it won't trigger a re-render if the availability changes).

Reply:
There is nothing in place that updates the reactivity. Only a page refresh as far as I know. So lets leave this in place like this untill it becomes a problem. cc @CGastrell do you know if the extension availabilty changes right away after purchasing the upgrade?